src/db_upgrade.c

```diff
@@ -1174,6 +1174,56 @@ static const struct db_upgrade_query db_upgrade_v2107_queries[] =
   };
 
 
+/* ---------------------------- 21.07 -> 21.08 ------------------------------ */
+
+#define U_v2108_ALTER_FILES_ADD_SOURCE \
+  "ALTER TABLE files ADD COLUMN source VARCHAR(255) DEFAULT NULL;"
+#define U_v2108_ALTER_PLAYLISTS_ADD_SOURCE \
+  "ALTER TABLE playlists ADD COLUMN source VARCHAR(255) DEFAULT NULL;"
+#define U_v2108_ALTER_DIR_ADD_SOURCE \
+  "ALTER TABLE directories ADD COLUMN source VARCHAR(255) DEFAULT NULL;"
+
+#define U_v2108_FILES_SET_SOURCE_RSS_SCANNER                     \
+  "UPDATE files SET source = 'rss-scanner' WHERE path in ("      \
+  "  SELECT i.filepath from playlists p, playlistitems i WHERE p.id = i.playlistid AND p.type = 4);"
+#define U_v2108_FILES_SET_SOURCE_SPOTIFY_SCANNER                    \
+  "UPDATE files SET source = 'spotify-scanner' WHERE virtual_path like '/spotify:/%';"
+#define U_v2108_FILES_SET_SOURCE_FILE_SCANNER                    \
+  "UPDATE files SET source = 'file-scanner' WHERE source IS NULL;"
+
+#define U_v2108_PL_SET_SOURCE_RSS_SCANNER                     \
+  "UPDATE playlists SET source = 'rss-scanner' WHERE type = 4;" // PL_RSS  = 4
+#define U_v2108_PL_SET_SOURCE_SPOTIFY_SCANNER                    \
+  "UPDATE playlists SET source = 'spotify-scanner' WHERE virtual_path like '/spotify:/%';"
+#define U_v2108_PL_SET_SOURCE_FILE_SCANNER                    \
+  "UPDATE playlists SET source = 'file-scanner' WHERE source IS NULL;"
+
+// Note: RSS feed items do not have their own directory structure (they use "http:/")
+#define U_v2108_DIR_SET_SOURCE_SPOTIFY_SCANNER                    \
+  "UPDATE directories SET source = 'spotify-scanner' WHERE virtual_path like '/spotify:/%';"
+#define U_v2108_DIR_SET_SOURCE_FILE_SCANNER                    \
+  "UPDATE directories SET source = 'file-scanner' WHERE virtual_path like '/file:/%';"
+
+#define U_v2108_SCVER_MINOR                    \
+  "UPDATE admin SET value = '08' WHERE key = 'schema_version_minor';"
```

Comment:
Here you are setting to "08", but above you had "8".

Reply:
We store the minor version with a leading zero in the DB.
The `SCHEMA_VERSION_MINOR` constant is a number that gets formatted in `db_init.c` to include a leading zero.
Defining it in the code with a leading zero only work up to `07`, because a leading zero here means the compiler should treat it as an octal number (thus 8 would be `010` and `08` throws a compiler error).

Reply:
Ah, that's tricky - well spotted!